### PR TITLE
Moving running scripts functionality to scripting library

### PR DESCRIPTION
**Pull request by jherico**
This PR enables low level support for removing the running scripts widget and replacing it with QML.

The new QML version is available for testing by running this script:  http://jherico.github.io/qmlscratch/runningScriptsTest.js

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5a57ee27887a3a46e260b482fa10d91cac368b41 available at: https://gist.github.com/97cbb49235da3e065dff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6800.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6800.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ba4b7f32fdea118f8500d1c0de0a1810ef20a15b available at: https://gist.github.com/97cbb49235da3e065dff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6800.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6800.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: ad3eed0ab895fde0c3da4d51f6a83bae57c2b681 available at: https://gist.github.com/97cbb49235da3e065dff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6800.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6800.dmg

**Comment by PhilipRosedale** [comment]
When I run http://jherico.github.io/qmlscratch/wireFrameTest.js

and bring up the menu, the QML window is just empty, with log errors: 

Done downloading script at: "http://jherico.github.io/qmlscratch/wireFrameTest.js"
[01/10 16:35:41] [DEBUG] Logging activity "loaded_script"
[01/10 16:35:41] [DEBUG] "[object Object]"
[01/10 16:35:41] [DEBUG] Created window with ID  4
[01/10 16:35:41] [WARNING] http://jherico.github.io/qmlscratch/qml/newUi/Main.qml:16:5: Type Hifi.FileDialog unavailable
[01/10 16:35:41] [WARNING] http://jherico.github.io/qmlscratch/qml/FileDialog.qml: File not found

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -3437,29 +3449,6 @@ bool Application::isHMDMode() const {
 }
 float Application::getTargetFrameRate() { return getActiveDisplayPlugin()->getTargetFrameRate(); }
 
-QRect Application::getDesirableApplicationGeometry() {
```
Comment (ZappoMan) [inline interface/src/Application.cpp]: Please don't remove this in this PR.
1) it has nothing to do with scripting
2) I want to get the Mac HMD working again, and this is clearly part of that old code path, so why delete it?

File: interface/src/Application.h [hunk 2]
```diff
@@ -499,12 +466,7 @@ private slots:
 
     TouchEvent _lastTouchEvent;
 
-    QReadWriteLock _scriptEnginesHashLock;
-    RunningScriptsWidget* _runningScriptsWidget;
-    QHash<QString, ScriptEngine*> _scriptEnginesHash;
-    bool _runningScriptsWidgetWasVisible;
-    QString _scriptsLocation;
-
+    RunningScriptsWidget* _runningScriptsWidget{ nullptr };
```
Comment (ZappoMan) [inline interface/src/Application.h]: coding standard space before '{'

File: interface/src/ui/RunningScriptsWidget.cpp [hunk 3]
```diff
@@ -172,7 +172,7 @@ void RunningScriptsWidget::showEvent(QShowEvent* event) {
         ui->filterLineEdit->setFocus();
     }
 
-    QRect parentGeometry = qApp->getDesirableApplicationGeometry();
```
Comment (ZappoMan) [inline interface/src/ui/RunningScriptsWidget.cpp]: please revert

File: libraries/script-engine/src/ScriptEngines.cpp [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+#endif
+
+ScriptsModel& getScriptsModel() {
+    static ScriptsModel scriptsModel;
+    return scriptsModel;
+}
+
+ScriptEngines::ScriptEngines()
+    : _scriptsLocationHandle("scriptsLocation", DESKTOP_LOCATION),
+      _previousScriptLocation("LastScriptLocation", DESKTOP_LOCATION){
```
Comment (ZappoMan) [inline libraries/script-engine/src/ScriptEngines.cpp]: coding standard says to place the `{` on the next line in 1st column for constructors

File: interface/src/Application.cpp (hunk 1 above)
Comment (jherico) [inline interface/src/Application.cpp]: prematurely removed.  restored

File: interface/src/Application.h (hunk 2 above)
Comment (jherico) [inline interface/src/Application.h]: fixed

File: interface/src/ui/RunningScriptsWidget.cpp (hunk 3 above)
Comment (jherico) [inline interface/src/ui/RunningScriptsWidget.cpp]: fixed

File: libraries/script-engine/src/ScriptEngines.cpp (hunk 4 above)
Comment (jherico) [inline libraries/script-engine/src/ScriptEngines.cpp]: fixed

**Comment by hifi-gustavo** [comment]
Build console output for commit: f3a1303a53577e8783e47e58cc66c7bd7b67da7b available at: https://gist.github.com/97cbb49235da3e065dff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6800.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6800.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f3a1303a53577e8783e47e58cc66c7bd7b67da7b available at: https://gist.github.com/97cbb49235da3e065dff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6800.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6800.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: d1d178787c07a1022a25bd983441519905463303 available at: https://gist.github.com/97cbb49235da3e065dff<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6800.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6800.dmg

**Pull request merged**